ICSharpCode.Decompiler/CSharp/ExpressionBuilder.cs

```diff
@@ -1901,7 +1901,16 @@ TranslatedExpression HandleCompoundAssignment(NumericCompoundAssign inst, Assign
 			if (inst.EvalMode == CompoundEvalMode.EvaluatesToOldValue)
 			{
 				Debug.Assert(op == AssignmentOperatorType.Add || op == AssignmentOperatorType.Subtract);
-				Debug.Assert(inst.Value.MatchLdcI(1) || inst.Value.MatchLdcF4(1) || inst.Value.MatchLdcF8(1));
+#if DEBUG
+				if (inst.Type is PointerType ptrType)
```

Comment:
```suggestion
				if (target.Type is PointerType ptrType)
```
`target.Type` is the actual C# type of the expression being incremented, so that's where it's important that the pointer element size matches.
I think `target.Type` and `inst.Type` will be equivalent in all the cases allowed by the compound assignment transform, so the changed assertion should still succeed.